packages/hash/shared/src/save.ts

```diff
@@ -345,6 +346,18 @@ const updateBlocks = defineOperation(
                     },
                   });
                 } else {
+                  if (!draftBlockData.accountId) {
```

Comment:
This works, but I feel we should check for the entityId instead. Makes it easier to reason about, especially since we majorly use entityId (and draftId) for most operations in the code.

Reply:
I don't know if the types are set up sophisticatedly enough - i.e, does TS know that if entityId exists, so does accountId, and vice-versa? I'll look into it.

Reply:
@teenoh Addressed this now, there's a new `isCreatedDraftEntity` type guard and types are constructed to know entityId and accountId are tied together

Reply:
Nice!